src/server/pps/server/api_server_test.go

```diff
@@ -405,3 +408,52 @@ func TestSetClusterDefaults(t *testing.T) {
 		require.True(t, defaults.CreatePipelineRequest.Autoscaling, "default autoscaling should be true after SetClusterDefaults")
 	})
 }
+
+func TestCreatePipelineV2(t *testing.T) {
+	ctx := pctx.TestContext(t)
+	env := realenv.NewRealEnv(ctx, t, dockertestenv.NewTestDBConfig(t))
+
+	_, err := env.PPSServer.SetClusterDefaults(ctx, &pps.SetClusterDefaultsRequest{
+		ClusterDefaultsJson: `{"create_pipeline_request": {"datum_tries": 17, "autoscaling": true}}`,
+	})
+	require.NoError(t, err, "SetClusterDefaults failed")
+
+	repo := "input"
+	pipeline := "pipeline"
+	require.NoError(t, env.PachClient.CreateRepo(pfs.DefaultProjectName, repo))
+	var pipelineTemplate = `{
+    "pipeline": {
+	"project": {
+	    "name": "{{.ProjectName | js}}"
+	},
+	"name": "{{.PipelineName | js}}"
+    },
+    "transform": {
+	"cmd": ["cp", "r", "/pfs/in", "/pfs/out"]
+    },
+    "input": {
+	"pfs": {
+	    "project": "default",
+	    "repo": "{{.RepoName | js}}",
+	    "glob": "/*",
+	    "name": "in"
+	}
+    },
+    "autoscaling": false
```

Comment:
Something that we need to test, or implement, are the two different forms of protojson key names.  A pipeline spec can use `snake_case` or `camelCase`.  So if the template is `{"some_key": "foo"}`, we need a test that `{"SomeKey": "bar"}` ends up as `{"SomeKey":"bar"}` and not `{"SomeKey":"bar","some_key":"foo"}`.  I believe running both things through protojson will end up with a canonical result (the `snake_case` type), but we should test it.  People, probably our own examples, have coupled to this protojson quirk.